src/state/queries/usePostThread/types.ts

```diff
@@ -0,0 +1,223 @@
+  | AppBskyUnspeccedGetPostThreadOtherV2.ThreadItem
+
+export const postThreadQueryKeyRoot = 'post-thread-v2' as const
+
+export const createPostThreadQueryKey = (props: PostThreadParams) =>
+  [postThreadQueryKeyRoot, props] as const
+
+export const createPostThreadOtherQueryKey = (
+  props: Omit<AppBskyUnspeccedGetPostThreadOtherV2.QueryParams, 'anchor'> & {
+    anchor?: string
+  },
+) => [postThreadQueryKeyRoot, 'other', props] as const
+
+export type PostThreadParams = Pick<
+  AppBskyUnspeccedGetPostThreadV2.QueryParams,
+  'sort' | 'prioritizeFollowedUsers'
+> & {
+  anchor?: string
+  view: 'tree' | 'linear'
+}
+
+export type UsePostThreadQueryResult = {
+  hasOtherReplies: boolean
+  thread: AppBskyUnspeccedGetPostThreadV2.ThreadItem[]
+  threadgate?: Omit<AppBskyFeedDefs.ThreadgateView, 'record'> & {
+    record: AppBskyFeedThreadgate.Record
+  }
+}
+
+export type ThreadItem =
+  | {
+      type: 'threadPost'
+      key: string
+      uri: string
+      depth: number
+      value: Omit<AppBskyUnspeccedDefs.ThreadItemPost, 'post'> & {
+        post: Omit<AppBskyFeedDefs.PostView, 'record'> & {
+          record: AppBskyFeedPost.Record
+        }
+      }
+      isBlurred: boolean
+      moderation: ModerationDecision
+      ui: {
+        isAnchor: boolean
+        showParentReplyLine: boolean
+        showChildReplyLine: boolean
+        indent: number
+        isLastChild: boolean
+        skippedIndentIndices: Set<number>
+        precedesChildReadMore: boolean
+      }
+    }
+  | {
+      type: 'threadPostNoUnauthenticated'
+      key: string
+      uri: string
+      depth: number
+      value: AppBskyUnspeccedDefs.ThreadItemNoUnauthenticated
+    }
+  | {
+      type: 'threadPostNotFound'
+      key: string
+      uri: string
+      depth: number
+      value: AppBskyUnspeccedDefs.ThreadItemNotFound
+    }
+  | {
+      type: 'threadPostBlocked'
+      key: string
+      uri: string
+      depth: number
+      value: AppBskyUnspeccedDefs.ThreadItemBlocked
+    }
+  | {
+      type: 'replyComposer'
+      key: string
+    }
+  | {
+      type: 'showOtherReplies'
+      key: string
+      onPress: () => void
+    }
+  | {
+      /*
+       * Read more replies, downwards in the thread.
+       */
+      type: 'readMore'
+      key: string
+      depth: number
+      href: string
+      moreReplies: number
+      skippedIndentIndices: Set<number>
+    }
+  | {
+      /*
+       * Read more parents, upwards in the thread.
+       */
+      type: 'readMoreUp'
+      key: string
+      href: string
+    }
+  | {
+      type: 'skeleton'
+      key: string
+      item: 'anchor' | 'reply' | 'replyComposer'
+    }
+
+/**
+ * Metadata collected while traversing the raw data from the thread response.
+ * Some values here can be computed immediately, while others need to be
+ * computed during a second pass over the thread after we know things like
+ * totaly number of replies, the reply index, etc.
```

Comment:
```suggestion
 * total number of replies, the reply index, etc.
```
little typo in comment